### PR TITLE
Add `no-async-fn-without-await` rule (fixes #132)

### LINKED ISSUE #132: `no-async-fn-without-await` rule
I've seen many tests where `async` functions are used without any `await` keyword. I'm pretty sure there's no reason to use `async` functions if you don't use `await` in it, right?

Example: https://github.com/sindresorhus/shell-env/pull/3/files/929775e4fbc79991d3fdfea40e6141ac4c64f211..fce68229a89e0d00fa1b703b752a2b9f70114375

Open to suggestions for a better rule name.

### PR DESCRIPTION
**Pull request by jfmengels**
Add `no-async-fn-without-await` rule (fixes #132)

### CODE REVIEW

File: docs/rules/no-async-fn-without-await.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Ensure that async tests use `await`
+
+Translations: [Français](https://github.com/avajs/ava-docs/blob/master/fr_FR/related/eslint-plugin-ava/docs/rules/no-async-fn-without-await.md)
+
+AVA comes with built-in support for [async functions](http://www.2ality.com/2016/02/async-functions.html) (async/await). This allows you to write shorter and clearer tests.
+
+Declaring an async test without using the `await` means that either a Promise is not awaited on as intended, or that the function could have be declared as a regular function, which is confusing.
```
Comment (sindresorhus) [inline docs/rules/no-async-fn-without-await.md]: > the `await` **keyword**

Comment (sindresorhus) [inline docs/rules/no-async-fn-without-await.md]: `be` => `been`

Comment (sindresorhus) [inline docs/rules/no-async-fn-without-await.md]: > which is confusing and slower.

File: docs/rules/no-async-fn-without-await.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+This rule will report an error when it finds an async test which does not use the `await` expression.
```
Comment (sindresorhus) [inline docs/rules/no-async-fn-without-await.md]: expression => keyword

**Comment by jfmengels** [comment]
Updated

**Comment by sindresorhus** [comment]
Perfect! We're now at 25 rules \o/

**Comment by jfmengels** [comment]
mQJe0rari2AIo

**Pull request merged**